rfcs/0001-repo-scaffold-tool.md

```diff
@@ -0,0 +1,112 @@
+
+###### Apply a template
+
+```bash
+npx @devcontainers/cli templates apply -t ghcr.io/pagopa/devcontainer-templates/node:1
+```
+
+###### Apply a template, specifying arguments
+
+```bash
+npx @devcontainers/cli templates apply \ 
+    --template-id ghcr.io/pagopa/devcontainer-templates/node:1 \ 
+    --template-args '{"nodeVersion": "20"}'
+```
+
+### Option 2 - Plop.js
+
+[Plop.js](https://plopjs.com/) is a code generator tool and framework that allows you to create custom generators, with complex logic, dynamic prompts and file manipulation.
+
+It is a Node.js based tool, that can be used as a command line tool or as a library (to be easily embedded in other JavaScript applications). It is also built-in into [Turborepo](https://turbo.build/docs/guides/generating-code), accessible by the `turbo gen` command.
+
+#### Authoring a generator (Plopfile)
+
+A Plop generator is defined in a `plopfile.js` file, that exports a function that receives a `plop` object. The `plop` object is used to define the generators and their prompts.
+
+A simple plopfile can look like this:
+
+```javascript
+export default function (plop) {
+    // controller generator
+    plop.setGenerator('controller', {
+        description: 'application controller logic',
+        prompts: [{
+            type: 'input',
+            name: 'name',
+            message: 'controller name please'
+        }],
+        actions: [{
+            type: 'add',
+            path: 'src/{{name}}.js',
+            templateFile: 'plop-templates/controller.hbs'
+        }]
+    });
+};
+```
+
+This generator will create a new file in the `src` folder, with the name specified by the user, using the template defined in `plop-templates/controller.hbs` (using [Handlebars](https://handlebarsjs.com/) as templating engine, as the extension suggests).
+
+Plop includes built-in `actions` for adding, modifying and deleting files, as well as for running custom scripts. It also supports custom actions, that can be used to extend the functionality.
+
+#### Using Plop to apply a template 
+
+Applying a template depends by how plop is used. If it is used as a command line tool, it can be run using the `plop` command. If it is used as a library, it can be run using its JavaScript API.
+
+###### Using Plop as a command line tool
+
+```bash
+npx plop
+# or
+npx turbo gen
+```
+
+###### Using Plop as a library
+
+https://plopjs.com/documentation/#wrapping-plop
+
+## Comparison
+
+| Feature | Development Container Templates | Plop.js |
+|---------|---------------------------------|---------|
+| File manipulation | Limited (copy only) | Advanced (add, modify, delete, run custom commands) |
+| Templating | Basic (variable substitution only) | Advanced (Handlebars) |
+| Authoring | Just a folder published to a registry | Should be distributed via NPM or alongside the consumer |
+
```

Comment:
I suggest to add the learning curve, as it could be steep when another tool is integrated: more dependencies, more things to know and maintain